src/translator/CMakeLists.txt

```diff
@@ -13,10 +13,15 @@ add_library(bergamot-translator STATIC
     sentence_ranges.cpp
     service.cpp
 )
+if (USE_WASM_COMPATIBLE_MARIAN)
+  # Using wasm compatible marian should include this compile definition;
+  # Has to be done here because of the marian's cmake practices
+  target_compile_definitions(bergamot-translator PUBLIC WASM)
+endif()
 
 if(COMPILE_WASM)
   # A dirty hack because of marian's bad cmake practices
```

Comment:
At this point, I'm inclined to say all the paths generated by CMake courtesy of this project gives more headhurt than marian's CMakeLists. Drop the native marian act here (why do you need stuff outside WASM with WASM_COMPATIBLE sources here anyway?), now that you've added workflow files in browsermt/marian-dev. There's only WASM or full marian.

Reply:
I suppose the useful part of this is being able to natively test a "reduced" Marian.  Though I'm not convinced it's really a reflective test given the changes on top to disable threads.  And I still don't understand why Mozilla keeps disabling threads when the plan is to use them.

Reply:
* What I want is some sanity in the build system configuration to be able to edit without much struggle. 
Therefore I want **only** two paths here - one which compiles to target wasm and one which compiles to full marian. Why are these being set in translator directory's CMakeLists, are these supposed to be set project-wide (in the root CMakeLists). I'm talking about the compile flags being added for WASM. I particularly have a problem with this after the exercise of `WITHOUT_PTHREADS` -> `__EMSCRIPTEN__` -> `WASM`. The build system ninja techniques are creating more work for everyone (at review, at edits).
* The two additional (github) workflows are unnecessary - as WASM builds are necessary and sufficient enough. It can't be magically such that WASM works and native breaks with WASM_COMPATIBLE sources. The bergamot-translator sources only have the two choices of WASM and not WASM, no WASM_COMPATIBLE_SOURCES mode specially.

Reply:
> Drop the native marian act here (why do you need stuff outside WASM with WASM_COMPATIBLE sources here anyway?)

I am sure that you know that compiling to wasm platform for inference requires a "reduced" marian. Having capability to compile "reduced" marian natively is helpful in debugging the wasm runtime failures by first reproducing the same failures on native platform with the "reduced" Marian binaries. In the past, this has proven to be useful where we could quickly diagnose a few wasm specific issues (e.g. `size_t` inconsistency).

As far as workflows are concerned, they are additional checks and are not hurting any one. Having these checks in Marian is not good enough because bergamot-translator is the final wasm artifact that is used in extension. We plan to keep these workflows. Your `full_marian` workflows are untouched. So it should not bother you anyway.